main/src/library/MutDisjointSet.flix

```diff
@@ -0,0 +1,149 @@
+/*
+ *  Copyright 2023 Jakob Schneider Villumsen
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+pub enum MutDisjointSet[t: Type, r: Eff]({
+    rc = Region[r],
+    forest = MutMap[t, MutDisjointSet.Node[t, r], r]
+})
+
+mod MutDisjointSet {
+
+    pub enum Node[t: Type, r: Eff]({
```

Comment:
Add comments.

Reply:
About each field.

Reply:
Are the field labels not self-explanatory? :thinking:

Reply:
Perhaps. Maybe not the rank field. I also thought we were dropping the Option and going for a self-loop, or is that not possible?

Reply:
Should be doable. I just thought `Option` was prefereable :)

Reply:
Actually maybe it does not work because we don't have null?

Reply:
Options are always preferable for data modelling, but for data structures I think all tricks count.

Reply:
That said, I am going to flipflop and suggest we stay with Option for now.